### ISSUE TITLE
Looking forward to call compensation functions immediately in case of one aborted event.

### ISSUE THREAD

**Issue by Gannalyo** [apache/incubator-servicecomb-saga#225]
So far, if Saga encountered an aborted event, then it could not execute compensation functions immediately yet. Instead, the Alpha EventScanner tool would call compensation methods within 500ms.

As we all know, 500ms is a long time in terms of computer. That will lead to a lot of wrong data probably. Hope to reduce the frequencies  on errors, even though we could not solve it completely at present.

![image](https://user-images.githubusercontent.com/34175244/42812478-5fe4dacc-89f0-11e8-9fa5-6909a441cfea.png)

Please point them out when I make mistakes. Thank you!

**Comment by WillemJiang** [comment]
Using the EventScanner is a challenge for us to handle the timeout event, we will do some modification here.  Since Saga only provides the eventual consistent,  we cannot guarantee that data is always right consistent here event we reduce the time to 1ms.

**Comment by Gannalyo** [comment]
Yeah, I understood. I just hope that we can make it be better. That's all.

**Comment by WillemJiang** [comment]
@Gannalyo  Can you close this issue if the answer address your question?

**Comment by WillemJiang** [comment]
Once the Alpha got the abort message ,  it knows about the transaction is failed and it can call the compensation method directly , the event scanner is just for the timeout event. 
I just fill a JIRA [SCB-835](https://issues.apache.org/jira/browse/SCB-835) for it.
